### PR TITLE
LABS-433 checkpoint and restore

### PR DESCRIPTION
**Pull request by Z-Shang**

### CODE REVIEW

File: jstests/stream/checkpoint.js [hunk 1]
```diff
@@ -33,8 +33,61 @@ function test1() {
...
+
+    // There should only be two checkpoints
+    // assert.eq(db.getSiblingDB("config").system.checkpoints.count(), 2);
+    // const checkpointDoc = db.getSiblingDB("config").system.checkpoints.findOne({});
+    // assert.eq(checkpointDoc["restoreCount"], 0);
+
+    // print("Checkpoint ID: ", checkpointDoc["_id"]);
+
+    // Wait for checkpoint markers to propagate
+    sleep(3000);
```
Comment (steve-1820) [inline jstests/stream/checkpoint.js:68]: @Z-Shang  What is this sleep for? There is another on line 58.

File: jstests/stream/manual_insertion_source.js [hunk 2]
```diff
@@ -16,8 +16,39 @@ function test1() {
...
 }
 
+function test2() {
+  try {
+    let agg = []
+
+    agg.push({ $window: { type: 'sliding', size: 20, gap: 20, unit: 'millisecond' } })
+
+    agg.push({
+      $groups: { _id: null, count: { $sum: 1 } }
```
Comment (michaelcahill) [inline jstests/stream/manual_insertion_source.js:26]: Can you please stick to the public API? One reason I'd like to make `$groups` and the other internal stages inaccessible so we don't keep adding code like this.

**Review by michaelcahill — APPROVED** [review]

File: jstests/stream/manual_insertion_source.js (hunk 2 above)
Comment (Z-Shang) [reply jstests/stream/manual_insertion_source.js:26]: Apparently the rewriting of stages is not yet implemented, I've created a ticket for it: https://jira.mongodb.org/browse/LABS-466

**Review by steve-1820 — APPROVED** [review]
LGTM

**Pull request merged**